x/programs/rust/wasmlanche_sdk/src/memory.rs

```diff
@@ -0,0 +1,92 @@
+//! Temporary storage allocated during the Program runtime.
+
+/// Represents a pointer to a block of memory allocated by the global allocator.
+pub struct Pointer {
+    raw: *mut u8,
+}
+
+impl Pointer {
+    #[must_use]
+    pub fn new(ptr: i64) -> Self {
+        Self {
+            raw: ptr as *mut u8,
+        }
+    }
+}
+
+/// Represents a block of memory allocated by the global allocator.
+pub struct Memory {
+    ptr: Pointer,
+}
+
+impl Memory {
+    #[must_use]
+    pub fn new(ptr: Pointer) -> Self {
+        Self { ptr }
+    }
+    /// Attempts return owned bytes from a pointer created by the global allocator.
+    ///
+    /// # Safety
+    /// `ptr` must be a pointer to a block of memory created using alloc.
+    /// `length` must be the length of the block of memory.
+    #[must_use]
+    pub unsafe fn range(&self, length: usize) -> Vec<u8> {
+        unsafe { Vec::from_raw_parts(self.ptr.raw, length, length) }
```

Comment:
hmm I think we don't want to own these bytes testing ` std::slice::from_raw_parts`

Reply:
added another variant range_owned with tests